app/models/participant_declaration.rb

```diff
@@ -89,7 +92,27 @@ def make_paid!
     DeclarationState.paid!(self) if payable?
   end
 
+  def make_ineligible!(reason: nil)
```

Comment:
```suggestion
  def make_ineligible!(reason:)
```

This one is up to you: whether we enforce a reason or leave it flexible.